modules/features/referrals/src/main/kotlin/au/com/shiftyjelly/pocketcasts/referrals/ReferralCardAnimatedGradientView.kt

```diff
@@ -0,0 +1,155 @@
+package au.com.shiftyjelly.pocketcasts.referrals
+
+import android.os.Build
+import androidx.compose.animation.core.AnimationVector2D
+import androidx.compose.animation.core.LinearOutSlowInEasing
+import androidx.compose.animation.core.RepeatMode
+import androidx.compose.animation.core.TwoWayConverter
+import androidx.compose.animation.core.animateValue
+import androidx.compose.animation.core.infiniteRepeatable
+import androidx.compose.animation.core.keyframes
+import androidx.compose.animation.core.rememberInfiniteTransition
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clipToBounds
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntOffset
+import au.com.shiftyjelly.pocketcasts.compose.extensions.rainbowBrush
+import au.com.shiftyjelly.pocketcasts.referrals.Position.BottomLeading
+import au.com.shiftyjelly.pocketcasts.referrals.Position.BottomTrailing
+import au.com.shiftyjelly.pocketcasts.referrals.Position.TopLeading
+import au.com.shiftyjelly.pocketcasts.referrals.Position.TopTrailing
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+
+private val positionVectors = listOf(
+    BottomLeading.vector,
+    TopLeading.vector,
+    TopTrailing.vector,
+    BottomTrailing.vector,
+)
+private val animationDuration = 5000L * positionVectors.size
+
+@Composable
+fun ReferralCardAnimatedBackgroundView(
+    modifier: Modifier = Modifier,
+) {
+    val infinityTransition = rememberInfiniteTransition(label = "infinite transition")
+    val animatePosition by infinityTransition.animateValue(
+        initialValue = positionVectors[0],
+        targetValue = positionVectors[1],
+        typeConverter = TwoWayConverter(
+            convertToVector = { position -> position },
+            convertFromVector = { vector ->
+                AnimationVector2D(vector.v1, vector.v2)
+            },
```

Comment:
I think you wanted initially to use `Position` as transition values and this got left over. 

```suggestion
            convertToVector = { vector -> vector },
            convertFromVector = { vector -> vector },
```

Reply:
🤔 looks like an issue with Compose. Let's leave it as is.